modules/typesniffer/typesniffer.go

```diff
@@ -81,6 +83,25 @@ func (ct SniffedType) GetMimeType() string {
 	return strings.SplitN(ct.contentType, ";", 2)[0]
 }
 
+func detectFileTypeBox(data []byte) (brands []string, found bool) {
+	if len(data) < 12 {
+		return nil, false
+	}
+	boxSize := int(binary.BigEndian.Uint32(data[:4]))
+	if boxSize < 12 || boxSize > len(data) {
+		return nil, false
+	}
+	tag := string(data[4:8])
+	if tag != "ftyp" {
+		return nil, false
+	}
+	brands = append(brands, string(data[8:12]))
+	for i := 16; i+4 <= boxSize; i += 4 {
+		brands = append(brands, string(data[i:i+4]))
+	}
+	return brands, true
+}
```

Comment:
Some comment what this does would be helpful.

Reply:
I think the code is clear enough, it is called file type box (ftyp)

Reply:
Add a link to some docs about it?

Reply:
google it ,  a lot : https://developer.apple.com/documentation/quicktime-file-format/file_type_compatibility_atom

But I guess we won't like to put an apple link here.

Reply:
How about a wikipedia link?

https://en.wikipedia.org/wiki/ISO_base_media_file_format#File_type_box